### PR TITLE
stats-7

### PR DESCRIPTION
**Pull request by CenkerAydin**

### CODE REVIEW

File: editor-components/stats/stats7/stats7.tsx [hunk 1]
```diff
@@ -2,103 +2,119 @@ import * as React from "react";
 import { BaseStats } from "../../EditorComponent";
 import styles from "./stats7.module.scss";
 
-type ICard = {
-  subtitle: string;
+
+interface Item {
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:6]:
```suggestion
type Item {
 title: JSX.Element;
 progress: number;
 progressText: JSX.Element;
}
```

[please check](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md)

File: editor-components/stats/stats7/stats7.tsx [hunk 2]
```diff
@@ -2,103 +2,119 @@ import * as React from "react";
...
-          displayer: "Card",
+          key: "item",
+          displayer: "Item",
           value: [
             {
               type: "string",
-              key: "subtitle",
-              displayer: "Subtitle",
-              value: "Products",
+              key: "progess-title",
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:48]: use camelCase for all keys. also keys must be same as you declared in type that you have written. ex:
```js
type Item {
 title: JSX.Element;
 progress: number;
 progressText: JSX.Element;
}
```

File: editor-components/stats/stats7/stats7.tsx [hunk 3]
```diff
@@ -107,34 +123,66 @@ class Stats7Page extends BaseStats {
...
-            <div className={this.decorateCSS("bottom-child")}>
-              {this.castToObject<ICard[]>("card-content").map(
-                (cardData: any, indexCard: number) => (
-                  <div key={`sts-7-${indexCard}`} className={this.decorateCSS("card")}>
-                    <h5 className={this.decorateCSS("card-data-subtitle")}>{cardData.subtitle}</h5>
-                    <h4 className={this.decorateCSS("card-data-title")}>{cardData.title}</h4>
-                    <p className={this.decorateCSS("card-data-description")}>{cardData.description}</p>
+        <div className={this.decorateCSS("max-content")}>
+
+          {subtitle && <div className={this.decorateCSS("title-child")}>
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:136]: for empty string check please read the documentation.

[read here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/stats/stats7/stats7.tsx [hunk 4]
```diff
@@ -107,34 +123,66 @@ class Stats7Page extends BaseStats {
...
+            {description && (
+              <p className={this.decorateCSS("description")}>
+                {description}
+              </p>
+            )}
+          </div>}
+
+          <div className={this.decorateCSS("progress-container")}>
+            {this.getPropValue("items").map((item: any, index: number) => {
+              const title = item.value[0].value;
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:152]: dont access props like this

File: editor-components/stats/stats7/stats7.tsx [hunk 5]
```diff
@@ -107,34 +123,66 @@ class Stats7Page extends BaseStats {
...
+            )}
+            {description && (
+              <p className={this.decorateCSS("description")}>
+                {description}
+              </p>
+            )}
+          </div>}
+
+          <div className={this.decorateCSS("progress-container")}>
+            {this.getPropValue("items").map((item: any, index: number) => {
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:151]: please cast to object. use the type you wrote.
```suggestion
            {this.castToObject<Item[]>("items").map((item: Item, index: number) => {
```

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle changes

File: editor-components/stats/stats7/stats7.tsx [hunk 6]
```diff
@@ -107,34 +123,74 @@ class Stats7Page extends BaseStats {
...
+                percent = 1;
+                text = "0%";
+              } else if (!text) {
+                text = `${percent}%`;
+              }
+              if(percent >= 100){
+                percent =100;
+              }
+            
+
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:167]:
```suggestion
if(this.castToString(title) || this.castToString(text))
```

File: editor-components/stats/stats7/stats7.tsx [hunk 7]
```diff
@@ -107,34 +123,74 @@ class Stats7Page extends BaseStats {
...
+            )}
+            {isDescriptionExist && (
+              <p className={this.decorateCSS("description")}>
+                {isDescriptionExist}
+              </p>
+            )}
+          </div>}
+
+          { itemList.length>0 && <div className={this.decorateCSS("progress-container")}>
+            {this.castToObject<Item[]>("items").map((item: any, index: number) => {
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:151]:
```suggestion
            {this.castToObject<Item[]>("items").map((item: Item, index: number) => {
```

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

File: editor-components/stats/stats7/stats7.tsx [hunk 8]
```diff
@@ -107,34 +123,73 @@ class Stats7Page extends BaseStats {
...
+              if(this.castToString(title) || this.castToString(progressText))
+              return (
+                <div className={this.decorateCSS("item")} key={index}>
+                  <div className={this.decorateCSS("progress-title")}>
+                    {title}
+                    <div className={this.decorateCSS("progress-percent")}>
+                      <div className={this.decorateCSS("progress-text")}>
+                        {text}
+                      </div>
+                    </div>
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:175]: don't render if {text} is empty

File: editor-components/stats/stats7/stats7.tsx [hunk 9]
```diff
@@ -107,34 +123,73 @@ class Stats7Page extends BaseStats {
...
                   </div>
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:176]: render if title or text is not empty

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes

File: editor-components/stats/stats7/stats7.module.scss [hunk 10]
```diff
@@ -1,95 +1,241 @@
...
+        padding: 4px 5px;
+        color: var(--composer-font-color-primary);
+        text-transform: uppercase;
+        font-weight: 600;
+      }
+
+      .title {
+        font-size: 2.83333rem;
+        line-height: 1.35294;
+        font-family: "Roboto", sans-serif;
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:32]: do not use font family

File: editor-components/stats/stats7/stats7.module.scss
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:1]: do not use font family and use px or % on margins

Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:1]: use height values in the height prop. do not use rem

File: editor-components/stats/stats7/stats7.tsx [hunk 11]
```diff
@@ -107,34 +123,78 @@ class Stats7Page extends BaseStats {
...
+            )}
+            {isDescriptionExist && (
+              <p className={this.decorateCSS("description")}>
+                {isDescriptionExist}
+              </p>
+            )}
+          </div>}
+
+          {itemList.length > 0 && <div className={this.decorateCSS("progress-container")}>
+            {this.castToObject<Item[]>("items").map((item: Item, index: number) => {
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:151]: declare this between redner and return statemens.
```javascript
const items = this.castToObject<Item[]>("items");
```
```suggestion
            {items.map((item: Item, index: number) => {
```

File: editor-components/stats/stats7/stats7.tsx [hunk 12]
```diff
@@ -107,34 +123,78 @@ class Stats7Page extends BaseStats {
...
+
+              if (percent === 0 || percent === null || percent === undefined) {
+                percent = 1;
+                text = "0%";
+              } else if (!text) {
+                text = `${percent}%`;
+              }
+              if (percent >= 100) {
+                percent = 100;
+              }
```
Comment (dogusmiuw) [inline editor-components/stats/stats7/stats7.tsx:164]: you may try to find a better way for this check

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

**Review by dogusmiuw — APPROVED** [review]

File: editor-components/stats/stats7/stats7.module.scss [hunk 13]
```diff
@@ -1,95 +1,222 @@
+* {
+  margin: 0;
+  padding: 0;
+}
+
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:5]: remove

File: editor-components/stats/stats7/stats7.tsx [hunk 14]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
-                  <div key={`sts-7-${indexCard}`} className={this.decorateCSS("card")}>
-                    <h5 className={this.decorateCSS("card-data-subtitle")}>{cardData.subtitle}</h5>
-                    <h4 className={this.decorateCSS("card-data-title")}>{cardData.title}</h4>
-                    <p className={this.decorateCSS("card-data-description")}>{cardData.description}</p>
+        <div className={this.decorateCSS("max-content")} style={{display: `${isTitleExist || isDescriptionExist ?   "flex":  "unset"}`}}>
+
+          {showDiv && <div className={this.decorateCSS("title-child")}>
+            {isSubtitleExist && (
+              <h1 className={this.decorateCSS("subTitle")} style={{
+                backgroundColor: `${enableSubtitleBackground ? "" : 'transparent' }`
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:146]: remove inline style

File: editor-components/stats/stats7/stats7.tsx [hunk 15]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+              }}>{isSubtitleExist}</h1>
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:147]: use getPropValue

File: editor-components/stats/stats7/stats7.tsx [hunk 16]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+            )}
+            {isTitleExist && (
+              <h1 className={this.decorateCSS("title")}>{isTitleExist}</h1>
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:150]: use getPropValue

File: editor-components/stats/stats7/stats7.tsx [hunk 17]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+            )}
+            {isDescriptionExist && (
+              <p className={this.decorateCSS("description")}>
+                {isDescriptionExist}
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:154]: use getPropValue

File: editor-components/stats/stats7/stats7.tsx [hunk 18]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+                percent = 100;
+              }
+
+              if (this.castToString(title) || this.castToString(progressText))
+                return (
+                  <div className={this.decorateCSS("item")} key={index}>
+
+                    {(title || text) && (
+                      <div className={this.decorateCSS("progress-title")}>
+                        {title}
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:178]: use getPropValue

File: editor-components/stats/stats7/stats7.tsx [hunk 19]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+                        {text && (
+                          <div className={this.decorateCSS("progress-percent")}>
+                            <div className={this.decorateCSS("progress-text")}>
+                              {text}
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:182]: use getPropValue

File: editor-components/stats/stats7/stats7.tsx [hunk 20]
```diff
@@ -107,34 +129,80 @@ class Stats7Page extends BaseStats {
...
+              } else if (percent >= 100) {
+                percent = 100;
+              }
+
+              if (this.castToString(title) || this.castToString(progressText))
+                return (
+                  <div className={this.decorateCSS("item")} key={index}>
+
+                    {(title || text) && (
+                      <div className={this.decorateCSS("progress-title")}>
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.tsx:177]: what if title not exists

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/stats/stats7/stats7.module.scss [hunk 21]
```diff
@@ -1,95 +1,333 @@
...
+        .progress-percent {
+          display: flex;
+          font-weight: bold;
+          margin-bottom: 5px;
+          color: (--composer-font-color-primary);
+        }
+      }
+
+      .progress-active {
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:193]: that's duplicate check line 88. remove one

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

**Review by Ayazelar — CHANGES_REQUESTED** [review]
remove duplicated styles on css side

File: editor-components/stats/stats7/stats7.module.scss [hunk 22]
```diff
@@ -1,95 +1,257 @@
...
+      align-items: end;
+    }
+
+    .progress-percent {
+      display: flex;
+      font-weight: bold;
+      margin-bottom: 5px;
+      color: (--composer-font-color-primary);
+      // width: 15%;
+      // justify-content: end;
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:80]: remove

File: editor-components/stats/stats7/stats7.module.scss [hunk 23]
```diff
@@ -1,95 +1,257 @@
...
+
+@media only screen and (max-width: 1200px) and (min-width:769px) {
+  .container {
+    padding: 20px 50px !important;
+
+    .max-content {
+      justify-content: space-between;
+
+      .title-child {
+        // gap: 20px;
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:144]:
```suggestion
```

File: editor-components/stats/stats7/stats7.module.scss [hunk 24]
```diff
@@ -1,95 +1,256 @@
...
+      align-items: end;
+    }
+
+    .progress-percent {
+      display: flex;
+      font-weight: bold;
+      margin-bottom: 5px;
+      color: (--composer-font-color-primary);
+      // width: 15%;
+      // justify-content: end;
```
Comment (Ayazelar) [inline editor-components/stats/stats7/stats7.module.scss:80]:
```suggestion
```

**Review by Ayazelar — APPROVED** [review]

**Pull request merged**